src/components/IFrame.js

```diff
@@ -100,17 +113,18 @@ export default class IFrame extends React.Component<Props, State> {
   };
 
   render(){
-    const {checkoutId, chargeId} = this.props;
+    const {checkoutId, chargeId, metadata} = this.props;
 
     const widgetType = checkoutId ? 'checkout' : chargeId ? 'charges' : null;
     if(!widgetType){
       throw new Error('must supply either checkoutId or chargeId prop');
     }
 
     const id = checkoutId ? checkoutId : chargeId;
+    const encodedMetadata = metadata ? this.encodeMetadata(metadata) : '';
 
     // $FlowFixMe Cannot call encodeURI with id bound to uri because undefined is incompatible with string.
-    const src = `${this.origin}/embed/${widgetType}/${encodeURI(id)}?origin=${encodeURI(this.hostName)}&version=${encodeURI(VERSION)}&buttonId=${this.uuid}`;
+    const src = `${this.origin}/embed/${widgetType}/${encodeURI(id)}?origin=${encodeURI(this.hostName)}&version=${encodeURI(VERSION)}&buttonId=${this.uuid}&custom=${encodedMetadata}`;
```

Comment:
might be worth stealin the query encoding function i wrote for checkout, or similar

```
    function encodeURIParams(params) {
      let encoded = []
      let quote = window.encodeURIComponent
      for (let key in params) {
        if (params.hasOwnProperty(key)) {
          encoded.push(quote(key) + '=' + quote(params[key]))
        }
      }
      return encoded.join('&')
    }
```